GoogleUtilities/Logger/Public/GoogleUtilities/GULLogger.h

```diff
@@ -68,6 +69,34 @@ extern BOOL GULIsLoggableLevel(GULLoggerLevel loggerLevel);
  */
 extern void GULLoggerRegisterVersion(NSString *version);
 
+/**
+ * Logs a message to the Xcode console and the device log. If running from AppStore, will
+ * not log any messages with a level higher than GULLoggerLevelNotice to avoid log spamming.
+ * (required) log level (one of the GULLoggerLevel enum values).
+ * (required) service name of type GULLoggerService.
+ * (required) message code starting with "I-" which means iOS, followed by a capitalized
+ *            three-character service identifier and a six digit integer message ID that is unique
+ *            within the service.
+ *            An example of the message code is @"I-COR000001".
+ * (required) message string which can be a format string.
+ * (optional) variable arguments list obtained from calling va_start, used when message is a format
+ *            string.
+ */
+extern void GULOSLogBasic(GULLoggerLevel level,
+                          NSString *subsystem,
+                          GULLoggerService category,
```

Comment:
Thoughts on removing the typedef in favor of `NSString *`?

Reply:
Paul and I both agree that switching to `NSString *` is less confusing. I haven't deleted the typedef in this PR since it's being used by internal projects but I added a deprecated comment so we can clean it up after. Also switched the new OSLog-based functions to `NSString *category`.